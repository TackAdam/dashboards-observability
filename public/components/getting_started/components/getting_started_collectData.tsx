/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState, useEffect } from 'react';
import {
  EuiAccordion,
  EuiPanel,
  EuiText,
  EuiSpacer,
  EuiCard,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSelect,
  EuiTabbedContent,
  EuiComboBox,
  EuiButton,
  EuiCodeBlock,
  EuiLink,
  EuiListGroup,
  EuiListGroupItem,
  EuiTitle,
} from '@elastic/eui';
import otelJson from './OTEL.json';

interface CollectAndShipDataProps {
  isOpen: boolean;
  onToggle: (isOpen: boolean) => void;
  selectedTechnology: string;
  onMoveToQueryData: () => void;
  onSelectSource: (source: string) => void;
}

export const CollectAndShipData: React.FC<CollectAndShipDataProps> = ({
  isOpen,
  onToggle,
  onMoveToQueryData,
  onSelectSource,
}) => {
  const [collectionMethod, setCollectionMethod] = useState('');
  const [specificMethod, setSpecificMethod] = useState('');
  const [gettingStarted, setGettingStarted] = useState<any>(null);
  const [selectedLabel, setSelectedLabel] = useState('');
  const [labelsOptions, setLabelsOptions] = useState([]);
  const [selectedTabId, setSelectedTabId] = useState('workflow_tab');
  const [saveMessage, setSaveMessage] = useState<string | null>(null);
  const [selectedWorkflow, setSelectedWorkflow] = useState('');

  useEffect(() => {
    if (selectedWorkflow && otelJson && otelJson['getting-started']) {
      const workflowData = otelJson['getting-started']?.workflows.find(
        (workflow) => workflow.name === selectedWorkflow
      );
      if (specificMethod === 'OTEL') {
        setGettingStarted(workflowData || null);
        const labels =
          workflowData?.steps?.map((step: any) => ({
            label: step.label || step.name,
            value: step.name.replace(/\s/g, '-'),
          })) || [];
        setLabelsOptions(labels);
      } else {
        setGettingStarted(null);
      }
    } else {
      setGettingStarted(null);
    }
  }, [selectedWorkflow, specificMethod]);

  const handleCollectionMethodChange = (value) => {
    setCollectionMethod(value);
    setSpecificMethod('');
    setSelectedWorkflow('');
    setGettingStarted(null);
  };

  const handleSpecificMethodChange = (change) => {
    const selectedMethod = change.target.value;
    setSpecificMethod(selectedMethod);
    onSelectSource(selectedMethod);
    setSelectedLabel('');
    setGettingStarted(null);
  };

  const handleWorkflowChange = (workflow) => {
    setSelectedWorkflow(workflow);
    setSelectedTabId('workflow_tab');
    const selectedWorkflowDataOTEL = otelJson['getting-started']?.workflows.find(
      (wf) => wf.name === workflow
    );
    if (specificMethod === 'OTEL') {
      setGettingStarted(selectedWorkflowDataOTEL || null);
    } else {
      setGettingStarted(null);
    }
    setSelectedLabel('');
  };

  const renderSpecificMethodDropdown = () => {
    if (!collectionMethod) return null;

    let options = [];
    if (collectionMethod === 'Configure collectors') {
      options = [
        { value: 'OTEL', text: 'Open Telemetry (structured)' },
        { value: 'Nginx', text: 'Nginx (structured)' },
        { value: 'vpc-flow', text: 'VPC Flow (structured)' },
        { value: 'Java', text: 'Java (unstructured)' },
        { value: 'Python', text: 'Python (unstructured)' },
        { value: 'golang', text: 'Golang (unstructured)' },
      ];
    } else if (collectionMethod === 'Upload a file CSV or JSON') {
      options = [
        { value: 'Fluent Bit', text: 'Fluent Bit' },
        { value: 'Data Prepper', text: 'Data Prepper' },
      ];
    }

    return (
      <>
        <EuiText>Select one of the following</EuiText>
        <EuiSpacer size="s" />
        <EuiSelect
          options={[{ value: '', text: 'Select an option' }, ...options]}
          value={specificMethod}
          onChange={handleSpecificMethodChange}
        />
      </>
    );
  };

  const onLabelChange = (selectedOptions: any[]) => {
    if (selectedOptions.length === 0) {
      setSelectedLabel('');
      window.scrollTo({
        top: 0,
        behavior: 'smooth',
      });
    } else {
      const selectedValue = selectedOptions[0].value;
      setSelectedLabel(selectedValue);
      const element = document.getElementById(selectedValue);
      if (element) {
        const headerOffset = 100;
        const elementPosition = element.getBoundingClientRect().top;
        const offsetPosition = elementPosition + window.pageYOffset - headerOffset;
        window.scrollTo({
          top: offsetPosition,
          behavior: 'smooth',
        });
      }
    }
  };

  const renderSteps = (workflow: any) => (
    <div>
      <EuiText>
        <h2>{workflow.name} Workflow</h2>
        <p>{workflow.description}</p>
        {workflow.info && (
          <EuiLink href={workflow.info} target="_blank">
            More Info
          </EuiLink>
        )}
        {workflow.types && <p>Types: {workflow.types.join(', ')}</p>}
      </EuiText>
      <EuiSpacer size="m" />
      {workflow.steps.map((step, index) => (
        <div key={index} id={step.name.replace(/\s/g, '-')}>
          <EuiTitle size="s">
            <h3>{step.name}</h3>
          </EuiTitle>
          {step.label && (
            <EuiText color="secondary">
              <p>{step.label}</p>
            </EuiText>
          )}
          <EuiText>{step.description}</EuiText>
          {step['input-params'] && step['input-params'].length > 0 && (
            <div>
              <EuiTitle size="xs">
                <h4>Input Parameters:</h4>
              </EuiTitle>
              {step['input-params'].map((param, idx) => (
                <EuiText key={idx}>
                  <strong>{param.name}:</strong> {param.description} ({param.type})
                </EuiText>
              ))}
            </div>
          )}
          {step.info &&
            step.info.map((link, linkIndex) => (
              <EuiLink key={linkIndex} href={link} target="_blank">
                More Info
              </EuiLink>
            ))}
          {step.content && (
            <EuiCodeBlock language="bash" fontSize="m" paddingSize="s" isCopyable>
              {step.content}
            </EuiCodeBlock>
          )}
          <EuiSpacer size="m" />
        </div>
      ))}
    </div>
  );

  const renderSchema = (schemas: any[]) =>
    schemas.map((schema, idx) => (
      <div key={idx}>
        <EuiTitle size="s">
          <h3>{schema.type} Schema</h3>
        </EuiTitle>
        <EuiText>
          {schema.description}
          <br />
          <strong>Alias:</strong> {schema.alias}
          <br />
          <strong>Index Pattern Name:</strong> {schema['index-pattern-name']}
          <br />
          {schema.info.map((infoLink: string, linkIdx: number) => (
            <EuiLink key={linkIdx} href={infoLink} target="_blank">
              More Info
            </EuiLink>
          ))}
        </EuiText>
        <EuiCodeBlock language="bash" fontSize="m" paddingSize="s" isCopyable>
          {schema.content}
        </EuiCodeBlock>
        <EuiLink href={schema['index-template']} target="_blank">
          Index Template
        </EuiLink>
        <EuiSpacer size="m" />
      </div>
    ));

  const renderIndex = (indexPatterns: any) => (
    <>
      <EuiText>
        {indexPatterns?.description}
        <br />
        {indexPatterns?.info?.map((infoLink: string, linkIdx: number) => (
          <EuiLink key={linkIdx} href={infoLink} target="_blank">
            More Info
          </EuiLink>
        ))}
      </EuiText>
      <EuiSpacer size="m" />
      <EuiTitle size="s">
        <h3>Index Patterns</h3>
      </EuiTitle>
      <EuiListGroup>
        {indexPatterns?.['index-patterns-name']?.map((pattern: string, idx: number) => (
          <EuiListGroupItem key={idx} label={pattern} />
        ))}
      </EuiListGroup>
      <EuiButton
        onClick={() => {
          setSaveMessage('Pattern created successfully');
          setTimeout(() => setSaveMessage(null), 3000);
        }}
      >
        Create Pattern
      </EuiButton>
      {saveMessage && (
        <EuiText color="secondary">
          <p>{saveMessage}</p>
        </EuiText>
      )}
      <EuiSpacer size="m" />
      <EuiButton onClick={onMoveToQueryData}>Move to Query and Analyze Data</EuiButton>
    </>
  );

  const tabs = [
    {
      id: 'workflow_tab',
      name: 'Workflow',
      content: (
        <div>
          <EuiComboBox
            placeholder="Search and select a label..."
            singleSelection={{ asPlainText: true }}
            options={labelsOptions}
            selectedOptions={selectedLabel ? [{ label: selectedLabel }] : []}
            onChange={onLabelChange}
            style={{ marginBottom: '20px' }}
          />
          {gettingStarted && renderSteps(gettingStarted)}
          <EuiButton
            onClick={() => {
              setSelectedTabId('schema_tab');
              window.scrollTo(0, 0);
            }}
          >
            Go to Schema
          </EuiButton>
        </div>
      ),
    },
    {
      id: 'schema_tab',
      name: 'Schema',
      content: (
        <div>
          <EuiTitle size="m">
            <h2>Schema</h2>
          </EuiTitle>
          {renderSchema(otelJson?.['getting-started']?.schema || [])}
          <EuiButton
            onClick={() => {
              setSelectedTabId('index_patterns_tab');
              window.scrollTo(0, 0);
            }}
          >
            Go to Index Patterns
          </EuiButton>
        </div>
      ),
    },
    {
      id: 'index_patterns_tab',
      name: 'Index Patterns',
      content: (
        <div>
          <EuiTitle size="m">
            <h2>Index Patterns</h2>
          </EuiTitle>
          {renderIndex(otelJson?.['getting-started']?.['index-patterns'] || {})}
        </div>
      ),
    },
  ];

  const onTabClick = (tab) => {
    setSelectedTabId(tab.id);
  };

  const renderContent = () => {
    if (!gettingStarted) {
      return (
        <EuiText>
          <h3>No workflow available</h3>
          <p>There is no workflow available for the selected method.</p>
        </EuiText>
      );
    }

    return (
      <EuiTabbedContent
        tabs={tabs}
        selectedTab={tabs.find((tab) => tab.id === selectedTabId)}
        onTabClick={onTabClick}
      />
    );
  };

  return (
    <EuiAccordion
      id="collect-and-ship-data"
      buttonContent="Collect & Ship Data"
      paddingSize="m"
      forceState={isOpen ? 'open' : 'closed'}
      onToggle={onToggle}
    >
      <EuiPanel>
        <EuiText>
          <h2>Collect Your Data</h2>
        </EuiText>
        <EuiSpacer size="m" />
        <EuiText>Select a collection method</EuiText>
        <EuiSpacer size="s" />
        <EuiFlexGroup gutterSize="l" style={{ maxWidth: '30rem' }}>
          <EuiFlexItem>
            <EuiCard
              layout="vertical"
              title="Configure collectors"
              description="Configure agents and ingestion pipeline"
              onClick={() => handleCollectionMethodChange('Configure collectors')}
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiCard
              layout="vertical"
              title="Upload a file CSV or JSON"
              description="..."
              onClick={() => handleCollectionMethodChange('Upload a file CSV or JSON')}
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiCard
              layout="vertical"
              title="Use a sample dataset"
              description="Explore with a log dataset"
              onClick={() => handleCollectionMethodChange('Use a sample dataset')}
            />
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiSpacer size="m" />
        {collectionMethod === 'Use a sample dataset' ? (
          <>
            <EuiText>Add all integration cards here.</EuiText>
          </>
        ) : (
          <>
            {renderSpecificMethodDropdown()}
            <EuiSpacer size="s" />
            {specificMethod && (
              <>
                <EuiSpacer size="s" />
                <EuiFlexGroup gutterSize="s" style={{ maxWidth: '20rem' }}>
                  <EuiFlexItem>
                    <EuiButton
                      fill={selectedWorkflow === 'QuickStart'}
                      onClick={() => handleWorkflowChange('QuickStart')}
                    >
                      QuickStart
                    </EuiButton>
                  </EuiFlexItem>
                  <EuiFlexItem>
                    <EuiButton
                      fill={selectedWorkflow === 'Connect To a Collector'}
                      onClick={() => handleWorkflowChange('Connect To a Collector')}
                    >
                      Connect to a Collector
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
              </>
            )}
            <EuiSpacer size="l" />
            {selectedWorkflow && renderContent()}
          </>
        )}
      </EuiPanel>
    </EuiAccordion>
  );
};
