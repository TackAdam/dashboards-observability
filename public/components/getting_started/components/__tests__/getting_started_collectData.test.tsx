/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { act } from 'react-dom/test-utils';
import { waitFor } from '@testing-library/react';
import { CollectAndShipData } from '../getting_started_collectData';

configure({ adapter: new Adapter() });

describe('Getting Started Collect Data Test', () => {
  const mockOnToggle = jest.fn();
  const mockOnMoveToQueryData = jest.fn();
  const mockOnSelectSource = jest.fn();
  const mockOnCardSelectionChange = jest.fn();

  const props = {
    isOpen: true,
    onToggle: mockOnToggle,
    selectedTechnology: '',
    onMoveToQueryData: mockOnMoveToQueryData,
    onSelectSource: mockOnSelectSource,
    onCardSelectionChange: mockOnCardSelectionChange,
  };

  it('Renders correctly', async () => {
    const wrapper = mount(<CollectAndShipData {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders collection method options', async () => {
    const wrapper = mount(<CollectAndShipData {...props} />);
    expect(wrapper.find('EuiCheckableCard')).toHaveLength(3);
    expect(wrapper.find('EuiCheckableCard').at(0).text()).toContain('Configure collectors');
    expect(wrapper.find('EuiCheckableCard').at(1).text()).toContain('Upload a file CSV');
    expect(wrapper.find('EuiCheckableCard').at(2).text()).toContain('Use a sample dataset');
  });

  it('Handles Configure collectors option', async () => {
    const wrapper = mount(<CollectAndShipData {...props} />);
    const configureCollectorsOption = wrapper.find('EuiCheckableCard').at(0).find('input');

    act(() => {
      configureCollectorsOption.simulate('change', { target: { checked: true } });
    });

    await waitFor(() => {
      wrapper.update();
      expect(wrapper.find('EuiSelectable')).toHaveLength(1);
      expect(wrapper.find('EuiSelectable').props().options).toEqual([
        { label: 'Open Telemetry (structured)', value: 'otel' },
        { label: 'Nginx (structured)', value: 'nginx' },
        { label: 'Java (unstructured)', value: 'java' },
        { label: 'Python (unstructured)', value: 'python' },
        { label: 'Golang (unstructured)', value: 'golang' },
      ]);
    });
  });

  it('Handles Upload a file CSV option', async () => {
    const wrapper = mount(<CollectAndShipData {...props} />);
    const uploadFileOption = wrapper.find('EuiCheckableCard').at(1).find('input');

    act(() => {
      uploadFileOption.simulate('change', { target: { checked: true } });
    });

    await waitFor(() => {
      wrapper.update();
      expect(wrapper.find('EuiSelectable')).toHaveLength(1);
      expect(wrapper.find('EuiSelectable').props().options).toEqual([
        { label: 'Fluent Bit', value: 'csv' },
      ]);
    });
  });

  it('Handles Use a sample dataset option', async () => {
    const wrapper = mount(<CollectAndShipData {...props} />);
    const sampleDatasetOption = wrapper.find('EuiCheckableCard').at(2).find('input');

    act(() => {
      sampleDatasetOption.simulate('change', { target: { checked: true } });
    });

    await waitFor(() => {
      wrapper.update();
      expect(wrapper.find('IntegrationCards')).toHaveLength(1);
    });
  });

  it('Selecting otel', async () => {
    const wrapper = mount(<CollectAndShipData {...props} />);
    const configureCollectorsOption = wrapper.find('EuiCheckableCard').at(0).find('input');

    act(() => {
      configureCollectorsOption.simulate('change', { target: { checked: true } });
    });

    await waitFor(() => {
      wrapper.update();
      expect(wrapper.find('EuiSelectable')).toHaveLength(1);
      expect(wrapper.find('EuiSelectable').props().options).toEqual([
        { label: 'Open Telemetry (structured)', value: 'otel' },
        { label: 'Nginx (structured)', value: 'nginx' },
        { label: 'Java (unstructured)', value: 'java' },
        { label: 'Python (unstructured)', value: 'python' },
        { label: 'Golang (unstructured)', value: 'golang' },
      ]);
    });

    const selectableProps = wrapper.find('EuiSelectable').props();
    const selectedOption = selectableProps.options?.find((opt) => opt.value === 'otel');

    if (selectableProps.onChange && selectedOption) {
      act(() => {
        selectableProps.onChange([{ ...selectedOption, checked: 'on' }] as any);
      });
    }

    await waitFor(() => {
      wrapper.update();
      expect(mockOnSelectSource).toHaveBeenCalledWith('otel');
      expect(wrapper.find('EuiTabbedContent')).toHaveLength(1);
      expect(wrapper.find('EuiSteps')).toHaveLength(1);
      expect(wrapper.find('EuiTitle').at(6).text()).toContain('Schema');
      expect(wrapper.find('EuiTitle').at(10).text()).toContain('Index Patterns');
    });
  });
});
