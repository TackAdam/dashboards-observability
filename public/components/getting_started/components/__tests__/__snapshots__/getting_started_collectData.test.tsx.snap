// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting Started Collect Data Test Renders correctly 1`] = `
<CollectAndShipData
  isOpen={true}
  onCardSelectionChange={[MockFunction]}
  onMoveToQueryData={[MockFunction]}
  onSelectSource={[MockFunction]}
  onToggle={[MockFunction]}
  selectedTechnology=""
>
  <EuiAccordion
    arrowDisplay="left"
    buttonContent="Collect and ship data"
    forceState="open"
    id="collect-and-ship-data"
    initialIsOpen={false}
    isLoading={false}
    isLoadingMessage={false}
    onToggle={[MockFunction]}
    paddingSize="m"
  >
    <div
      className="euiAccordion euiAccordion-isOpen"
      onToggle={[MockFunction]}
    >
      <div
        className="euiAccordion__triggerWrapper"
      >
        <button
          aria-controls="collect-and-ship-data"
          aria-expanded={true}
          className="euiAccordion__button"
          id="random_html_id"
          onClick={[Function]}
          type="button"
        >
          <span
            className="euiAccordion__iconWrapper"
          >
            <EuiIcon
              className="euiAccordion__icon euiAccordion__icon-isOpen"
              size="m"
              type="arrowRight"
            >
              <EuiIconBeaker
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon euiAccordion__icon-isOpen"
                focusable="false"
                role="img"
                style={null}
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon euiAccordion__icon-isOpen"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                  />
                </svg>
              </EuiIconBeaker>
            </EuiIcon>
          </span>
          <span
            className="euiIEFlexWrapFix"
          >
            Collect and ship data
          </span>
        </button>
      </div>
      <div
        aria-labelledby="random_html_id"
        className="euiAccordion__childWrapper"
        id="collect-and-ship-data"
        role="region"
        tabIndex={-1}
      >
        <EuiResizeObserver
          onResize={[Function]}
        >
          <div>
            <div
              className="euiAccordion__padding--m"
            >
              <EuiPanel>
                <div
                  className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
                >
                  <EuiText>
                    <div
                      className="euiText euiText--medium"
                    >
                      <h3>
                        Collect your data
                      </h3>
                    </div>
                  </EuiText>
                  <EuiSpacer
                    size="m"
                  >
                    <div
                      className="euiSpacer euiSpacer--m"
                    />
                  </EuiSpacer>
                  <EuiText>
                    <div
                      className="euiText euiText--medium"
                    >
                      <strong>
                        Select a collection method
                      </strong>
                    </div>
                  </EuiText>
                  <EuiSpacer
                    size="s"
                  >
                    <div
                      className="euiSpacer euiSpacer--s"
                    />
                  </EuiSpacer>
                  <EuiFlexGroup>
                    <div
                      className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                    >
                      <EuiFlexItem>
                        <div
                          className="euiFlexItem"
                        >
                          <EuiCheckableCard
                            checkableType="radio"
                            checked={false}
                            id="configure_collectors"
                            label="Configure collectors"
                            onChange={[Function]}
                          >
                            <_EuiSplitPanelOuter
                              className="euiCheckableCard"
                              direction="row"
                              hasBorder={true}
                              responsive={false}
                            >
                              <EuiPanel
                                className="euiSplitPanel euiSplitPanel--row euiCheckableCard"
                                grow={false}
                                hasBorder={true}
                                paddingSize="none"
                              >
                                <div
                                  className="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
                                >
                                  <_EuiSplitPanelInner
                                    color="subdued"
                                    grow={false}
                                    onClick={[Function]}
                                  >
                                    <EuiPanel
                                      borderRadius="none"
                                      className="euiSplitPanel__inner"
                                      color="subdued"
                                      element="div"
                                      grow={false}
                                      hasBorder={false}
                                      hasShadow={false}
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                                        onClick={[Function]}
                                      >
                                        <EuiRadio
                                          checked={false}
                                          id="configure_collectors"
                                          onChange={[Function]}
                                        >
                                          <div
                                            className="euiRadio euiRadio--noLabel"
                                          >
                                            <input
                                              checked={false}
                                              className="euiRadio__input"
                                              id="configure_collectors"
                                              onChange={[Function]}
                                              type="radio"
                                            />
                                            <div
                                              className="euiRadio__circle"
                                            />
                                          </div>
                                        </EuiRadio>
                                      </div>
                                    </EuiPanel>
                                  </_EuiSplitPanelInner>
                                  <_EuiSplitPanelInner>
                                    <EuiPanel
                                      borderRadius="none"
                                      className="euiSplitPanel__inner"
                                      color="transparent"
                                      element="div"
                                      hasBorder={false}
                                      hasShadow={false}
                                    >
                                      <div
                                        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                                      >
                                        <label
                                          aria-describedby="configure_collectors-details"
                                          className="euiCheckableCard__label"
                                          htmlFor="configure_collectors"
                                        >
                                          Configure collectors
                                        </label>
                                        <div
                                          className="euiCheckableCard__children"
                                          id="configure_collectors-details"
                                        >
                                          Configure agents and ingestion pipeline
                                        </div>
                                      </div>
                                    </EuiPanel>
                                  </_EuiSplitPanelInner>
                                </div>
                              </EuiPanel>
                            </_EuiSplitPanelOuter>
                          </EuiCheckableCard>
                        </div>
                      </EuiFlexItem>
                      <EuiFlexItem>
                        <div
                          className="euiFlexItem"
                        >
                          <EuiCheckableCard
                            checkableType="radio"
                            checked={false}
                            id="upload_file"
                            label="Upload a file CSV"
                            onChange={[Function]}
                          >
                            <_EuiSplitPanelOuter
                              className="euiCheckableCard"
                              direction="row"
                              hasBorder={true}
                              responsive={false}
                            >
                              <EuiPanel
                                className="euiSplitPanel euiSplitPanel--row euiCheckableCard"
                                grow={false}
                                hasBorder={true}
                                paddingSize="none"
                              >
                                <div
                                  className="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
                                >
                                  <_EuiSplitPanelInner
                                    color="subdued"
                                    grow={false}
                                    onClick={[Function]}
                                  >
                                    <EuiPanel
                                      borderRadius="none"
                                      className="euiSplitPanel__inner"
                                      color="subdued"
                                      element="div"
                                      grow={false}
                                      hasBorder={false}
                                      hasShadow={false}
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                                        onClick={[Function]}
                                      >
                                        <EuiRadio
                                          checked={false}
                                          id="upload_file"
                                          onChange={[Function]}
                                        >
                                          <div
                                            className="euiRadio euiRadio--noLabel"
                                          >
                                            <input
                                              checked={false}
                                              className="euiRadio__input"
                                              id="upload_file"
                                              onChange={[Function]}
                                              type="radio"
                                            />
                                            <div
                                              className="euiRadio__circle"
                                            />
                                          </div>
                                        </EuiRadio>
                                      </div>
                                    </EuiPanel>
                                  </_EuiSplitPanelInner>
                                  <_EuiSplitPanelInner>
                                    <EuiPanel
                                      borderRadius="none"
                                      className="euiSplitPanel__inner"
                                      color="transparent"
                                      element="div"
                                      hasBorder={false}
                                      hasShadow={false}
                                    >
                                      <div
                                        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                                      >
                                        <label
                                          aria-describedby="upload_file-details"
                                          className="euiCheckableCard__label"
                                          htmlFor="upload_file"
                                        >
                                          Upload a file CSV
                                        </label>
                                        <div
                                          className="euiCheckableCard__children"
                                          id="upload_file-details"
                                        >
                                          Upload your data
                                        </div>
                                      </div>
                                    </EuiPanel>
                                  </_EuiSplitPanelInner>
                                </div>
                              </EuiPanel>
                            </_EuiSplitPanelOuter>
                          </EuiCheckableCard>
                        </div>
                      </EuiFlexItem>
                      <EuiFlexItem>
                        <div
                          className="euiFlexItem"
                        >
                          <EuiCheckableCard
                            checkableType="radio"
                            checked={false}
                            id="use_sample_dataset"
                            label="Use a sample dataset"
                            onChange={[Function]}
                          >
                            <_EuiSplitPanelOuter
                              className="euiCheckableCard"
                              direction="row"
                              hasBorder={true}
                              responsive={false}
                            >
                              <EuiPanel
                                className="euiSplitPanel euiSplitPanel--row euiCheckableCard"
                                grow={false}
                                hasBorder={true}
                                paddingSize="none"
                              >
                                <div
                                  className="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
                                >
                                  <_EuiSplitPanelInner
                                    color="subdued"
                                    grow={false}
                                    onClick={[Function]}
                                  >
                                    <EuiPanel
                                      borderRadius="none"
                                      className="euiSplitPanel__inner"
                                      color="subdued"
                                      element="div"
                                      grow={false}
                                      hasBorder={false}
                                      hasShadow={false}
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                                        onClick={[Function]}
                                      >
                                        <EuiRadio
                                          checked={false}
                                          id="use_sample_dataset"
                                          onChange={[Function]}
                                        >
                                          <div
                                            className="euiRadio euiRadio--noLabel"
                                          >
                                            <input
                                              checked={false}
                                              className="euiRadio__input"
                                              id="use_sample_dataset"
                                              onChange={[Function]}
                                              type="radio"
                                            />
                                            <div
                                              className="euiRadio__circle"
                                            />
                                          </div>
                                        </EuiRadio>
                                      </div>
                                    </EuiPanel>
                                  </_EuiSplitPanelInner>
                                  <_EuiSplitPanelInner>
                                    <EuiPanel
                                      borderRadius="none"
                                      className="euiSplitPanel__inner"
                                      color="transparent"
                                      element="div"
                                      hasBorder={false}
                                      hasShadow={false}
                                    >
                                      <div
                                        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                                      >
                                        <label
                                          aria-describedby="use_sample_dataset-details"
                                          className="euiCheckableCard__label"
                                          htmlFor="use_sample_dataset"
                                        >
                                          Use a sample dataset
                                        </label>
                                        <div
                                          className="euiCheckableCard__children"
                                          id="use_sample_dataset-details"
                                        >
                                          Explore with a log dataset
                                        </div>
                                      </div>
                                    </EuiPanel>
                                  </_EuiSplitPanelInner>
                                </div>
                              </EuiPanel>
                            </_EuiSplitPanelOuter>
                          </EuiCheckableCard>
                        </div>
                      </EuiFlexItem>
                    </div>
                  </EuiFlexGroup>
                  <EuiSpacer
                    size="m"
                  >
                    <div
                      className="euiSpacer euiSpacer--m"
                    />
                  </EuiSpacer>
                  <EuiSpacer
                    size="s"
                  >
                    <div
                      className="euiSpacer euiSpacer--s"
                    />
                  </EuiSpacer>
                </div>
              </EuiPanel>
            </div>
          </div>
        </EuiResizeObserver>
      </div>
    </div>
  </EuiAccordion>
</CollectAndShipData>
`;
